### PR TITLE
Fix global default http proxy lookup

### PR DESCRIPTION
**Pull request by tpapaioa**
This PR fixes the logic for looking up the global default http proxy for content repositories. In Satellite 6.8, the setting has a value of 'no global default' when unset, while on Satellite 6.9 the value is 'Empty'.

### CODE REVIEW

**Review by swadeley — APPROVED** [review]
ACK, pending test result.

File: airgun/entities/repository.py [hunk 1]
```diff
@@ -14,20 +14,24 @@
...
-    def _find_default_http_proxy(self, values):
-        """ Finds the global proxy name and returns updated values accordingly."""
-        proxy = SettingsEntity(self.browser)
-        result = proxy.read(property_name='name = content_default_http_proxy')['table'][0]['Value']
-        global_proxy_name = result.split()[0]
-        if global_proxy_name == 'no':
-            global_proxy_name = 'None'
-        values["repo_content.http_proxy_policy"] = f'Global Default ({global_proxy_name})'
-        return values
+    def _global_default_http_proxy(self):
```
Comment (mshriver) [inline airgun/entities/repository.py:17]: Consider making this a property.

File: airgun/entities/repository.py [hunk 2]
```diff
@@ -14,20 +14,24 @@
...
+        """Look up the default http proxy and return the string that a user would select for
+        HTTP Proxy Policy when creating or updating a repository.
+        """
+        proxy_name = SettingsEntity(self.browser).read(
+            property_name='name = content_default_http_proxy'
+        )['table'][0]['Value']
+
+        # The default text for no default http proxy varies between versions of Satellite
+        if proxy_name in ('Empty', 'no global default'):
+            proxy_name = 'None'
```
Comment (mshriver) [inline airgun/entities/repository.py:27]: Not really related to the change you're making as this is the same in the old function... but a _string_ `'None'`?? *WHY???*

**Review by mshriver — APPROVED** [review]

File: airgun/entities/repository.py (hunk 2 above)
Comment (tpapaioa) [reply airgun/entities/repository.py:27]: That's how it shows up in the dropdown selector if no global default http proxy has been configured: 'Global Default (None)'.

**Comment by tpapaioa** [comment]
Test results:

```
# pytest tests/foreman/ui/test_http_proxy.py -k test_positive_assign_http_proxy_to_products_repositories
[...]
collected 2 items / 1 deselected / 1 selected                                                                                                                                                                                                

tests/foreman/ui/test_http_proxy.py . [100%]

[...]
=== 1 passed, 1 deselected in 1258.55s (0:20:58) ===
```

File: airgun/entities/repository.py (hunk 1 above)
Comment (tpapaioa) [reply airgun/entities/repository.py]: Done.

**Pull request merged**